jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -0,0 +1,132 @@
+from astropy.nddata import NDDataArray, StdDevUncertainty
+from astropy.wcs.wcsapi.fitswcs import SlicedFITSWCS
+from glue_astronomy.translators.spectrum1d import PaddedSpectrumWCS
+from traitlets import List, Unicode
+
+from jdaviz.core.events import SnackbarMessage
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import (PluginTemplateMixin,
+                                        SelectPluginComponent,
+                                        SpatialSubsetSelectMixin,
+                                        AddResultsMixin)
+from jdaviz.core.user_api import PluginUserApi
+
+__all__ = ['SpectralExtraction']
+
+
+@tray_registry(
+    'cubeviz-spectral-extraction', label="Spectral Extraction", viewer_requirements='spectrum'
+)
+class SpectralExtraction(PluginTemplateMixin, SpatialSubsetSelectMixin, AddResultsMixin):
```

Comment:
`AddResultsMixin` should have a default label assigned somewhere so you don't always need to provide it.  You can also watch other traitlets to change the default (see implementations in `GaussianSmooth._set_default_results_label`, etc)... in this case the dataset is fixed, but maybe would want to have it depend on the name of the selected spatial subset and function?